Automattic/jetpack
# Paid blocks: Improve rendering of the upgrade nudge

Fixes https://github.com/Automattic/jetpack/issues/25102

#### Changes proposed in this Pull Request:

This PR adds a couple of enhancements to the paid blocks upgrade banner displayed on WP.com sites:

- Places the nudge inside the actual block edit component. Previously it was a sibling of that component, forcing us to hack CSS of the block toolbar so it doesn't overlap the nudge. Now that the nudge is part of the block edit markup, the toolbar is positioned correctly, effectively allowing us to remove the fixed height hack that was causing issues for cases when there is not enough width.

Before | After
--- | ---
<img width="645" alt="Screen Shot 2022-08-29 at 11 56 52" src="https://user-images.githubusercontent.com/1233880/187177193-f0104836-1a50-4df5-865d-ddc7ba911ef5.png"> | <img width="669" alt="Screen Shot 2022-08-29 at 11 57 41" src="https://user-images.githubusercontent.com/1233880/187177213-b74ec56a-fdd1-4aee-9240-cbed63087dde.png">

- Hides the nudge of nested paid blocks, and ensures that the the one of the parent block is the only one visible.

Before | After
--- | ---
<img width="647" alt="Screen Shot 2022-08-25 at 12 53 48" src="https://user-images.githubusercontent.com/1233880/186646749-bd54585f-7ca4-4d48-8388-c8f6764e26ea.png"> | <img width="641" alt="Screen Shot 2022-08-25 at 12 54 31" src="https://user-images.githubusercontent.com/1233880/186646761-78d15edf-08b3-4cdb-8f26-dd1d4b891483.png">


#### Other information:


#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Apply the diff created by Fusion to your WP.com sandbox.
- Sandbox a WP.com site.
- Go to https://wordpress.com/post.
- Select a Free site.
- Insert a Columns block with a 33 / 33 / 33 layout.
- Insert any paid block inside one of the columns.
- Make sure the upgrade nudge is displayed correctly with its height adapted to the content of the nudge.
- Insert a paid block in the root of the editor that contains another inner paid block. Some examples:
  - Payment buttons.
  - WhatApp button.
  - Premium content with a video in the subscriber content.
- Make sure that the upgrade nudger of the parent block is the only one that is visible.
